blob/azureblob/example_test.go

```diff
@@ -25,74 +29,215 @@ import (
-	key := "my-file.json"
 
-	// Initializes an azblob.ServiceURL which represents the Azure Storage Account using Shared Key authorization
-	serviceURL, _ := azureblob.ServiceURLFromAccountKey(accountName, accountKey)
+	// Initializes an azblob.ServiceURL which represents the Azure Storage Account using Shared Key authorization.
+	serviceURL, err := azureblob.ServiceURLFromAccountKey(accountName, accountKey)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	// Credential represents the authorizer for SignedURL.
+	creds, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
 
-	// Credential represents the authorizer for SignedURL
-	creds, _ := azblob.NewSharedKeyCredential(accountName, accountKey)
+	// Override the default retry policy so calls can return promptly for this test. Please review the timeout guidelines and set accordingly.
+	// See https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations for more information.
+	opts := azblob.PipelineOptions{
+		Retry: azblob.RetryOptions{
+			Policy:        azblob.RetryPolicyFixed,
+			TryTimeout:    10 * time.Second,
+			MaxTries:      1,
+			RetryDelay:    0 * time.Second,
+			MaxRetryDelay: 0 * time.Second,
+		},
+	}
+	p := azblob.NewPipeline(creds, opts)
+	*serviceURL = serviceURL.WithPipeline(p)
+
+	if err != nil {
+		// This is expected since accountName and accountKey is empty.
+		log.Fatal(err)
+		return
+	}
+
+	// Set credential for the SignedURL operation.
 	azureOpts := azureblob.Options{
 		Credential: creds,
 	}
 
-	// Creates a *blob.Bucket backed by Azure Storage Account
-	b, _ := azureblob.OpenBucket(ctx, serviceURL, bucketName, &azureOpts)
+	// Create a *blob.Bucket.
+	b, err := azureblob.OpenBucket(ctx, serviceURL, bucketName, &azureOpts)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	_, err = b.ReadAll(ctx, "my-key")
+	if err != nil {
+		// This is expected due to the fake credentials we used above.
+		fmt.Println("ReadAll failed due to invalid credentials")
+	}
+
+	// Output:
+	// ReadAll failed due to invalid credentials
+}
+
+func Example_open() {
+
+	// Open creates a *Bucket from a URL.
+	// Example URL: azblob://my-bucket?cred_path=replace-with-path-to-credentials-file
+	// The schema "azblob" represents the provider for Azure Blob (Azure Storage Account).
+	// The host "my-bucket" represents the blob bucket (Azure Storage Account's Container).
+	// The queryString "cred_path: represents the credentials file in JSON format. See the package documentation for credentials file schema.
+
+	_, err := blob.Open(context.Background(), "azblob://my-bucket?cred_path=replace-with-path-to-credentials-file")
+	if err != nil {
+		// Windows will return error: replace-with-path-to-credentials-file: The system cannot find the file specified.
+		// macOS and Linux will return error: open replace-with-path-to-credentials-file: no such file or directory
+		fmt.Println(err)
```

Comment:
Don't print the specific error, just print something like "blob.Open failed", that way it's not dependent on the OS or Go version.

Reply:
Done.